tests/test_activate.py

```diff
@@ -1,652 +1,582 @@
+
+else:
+    # Only run these tests for commands that are installed.
+    shells = ['bash', 'zsh']
+    ps_var = "PS1"
+    var_format = "${var}"
+    binpath = "/bin/"  # mind the trailing slash.
+    source_setup = "source"
+    nul = '2>/dev/null'
+    set_var = 'export '
+    shell_suffix = ""
+    printps1 = '{echo} {var}'.format(echo=echo, var=var_format.format(var=ps_var))
+
+for shell in shells[:]:
+    try:
+        stdout, stderr = run_in('echo', shell)
+    except OSError:
+        pass
+    else:
+        if not stderr:
+            shells.append(shell)
+
+raw_ps = escape_curly(os.getenv(ps_var, ""))
+def print_ps1(env_dirs, number):
+    return u" ".join([u"({})".format(os.path.split(env_dirs[number])[-1]), raw_ps]).strip()
+
+CONDA_ENTRY_POINT = """\
 #!{syspath}/python
 import sys
 from conda.cli import main
 
 sys.exit(main())
 """
+printpath = '{echo} {var}'.format(echo=echo, var=var_format.format(var="PATH"))
+if sys.platform == "win32":
+    printdefaultenv = '{echo}.{var}'.format(echo=echo, var=var_format.format(var="CONDA_ACTIVE_ENV"))
+else:
+    printdefaultenv = '{echo} {var}'.format(echo=echo, var=var_format.format(var="CONDA_ACTIVE_ENV"))
 
 command_setup = """\
-export PATH="{ROOTPATH}"
-export PS1='$'
-export PYTHONPATH="{PYTHONPATH}"
-export CONDARC=' '
-cd {here}
-""".format(here=dirname(__file__), ROOTPATH=ROOTPATH, PYTHONPATH=PYTHONPATH)
-
-command_setup = command_setup + """
-mkdir -p {envs}/test1/bin
-mkdir -p {envs}/test2/bin
-"""
-
+set {ps_var}={raw_ps}
+set PYTHONPATH={PYTHONPATH}
+set CONDARC=
+""".format(here=dirname(__file__), PYTHONPATH=PYTHONPATH,
+           ps_var=ps_var, raw_ps=raw_ps)
+
+if platform.startswith("win"):
+    command_setup = "@echo off\n" + command_setup
+
+
+_format_vars = {
+    'nul': nul,
+    'printpath': printpath,
+    'printdefaultenv': printdefaultenv,
+    'printps1': printps1,
+    'set_var': set_var,
+    'source': source_setup,
+    'binpath': binpath,
+    'syspath': sys.prefix,
+    'shell_suffix': shell_suffix
+}
 
 @pytest.mark.slow
 def test_activate_test1():
     for shell in shells:
         with TemporaryDirectory(prefix='envs', dir=dirname(__file__)) as envs:
-            activate, deactivate, conda = _write_entry_points(envs)
             commands = (command_setup + """
-            source {activate} {envs}/test1
-            printf $PATH
-            """).format(envs=envs, activate=activate)
+            {source} {syspath}{binpath}activate{shell_suffix} "{env_dirs[0]}"
+            {printpath}
+            """).format(envs=envs, env_dirs=gen_test_env_paths(envs), **_format_vars)
 
             stdout, stderr = run_in(commands, shell)
-            assert stdout == envs + "/test1/bin:" + PATH
-            assert stderr == 'discarding {syspath} from PATH\nprepending {envs}/test1/bin to PATH\n'.format(envs=envs, syspath=syspath)
+            assert_equals(stderr, u'prepending {envpaths} to PATH'\
```

Comment:
I think this is hard to solve generally.  The way that I have things now, I consider the whole block of additional paths as one item.  If I need to go find the root env, I have rather large misgivings that users may have changed their PATH themselves, and totally trashed my assumption that I can treat the collection of paths as a single entity for purposes of string replacement.  If that assumption is violated, then searching the path for each piece individually, and re-creating the PATH faithfully afterwards, is a mundane but difficult bookkeeping task.  I could just store the original PATH when any environment is activated, I guess.

This will require quite a bit more work to do right, and I'd like to punt.  I would like to create another issue for this, and solve it soon.  If you have time, I would welcome any effort you can contribute towards this.